pySDC/implementations/problem_classes/generic_spectral.py

```diff
@@ -0,0 +1,307 @@
+        self.M = self._setup_operator(LHS)
+
+    def setup_preconditioner(self, Dirichlet_recombination=True, left_preconditioner=True):
+        """
+        Get left and right precondioners. A right preconditioner of D2T will result in Dirichlet recombination. 10/10 would recommend!
+
+        Args:
+            Dirichlet_recombination (bool): Basis conversion for right precondioner
+            left_preconditioner (bool): If True, it will interleave the variables and reverse the Kronecker product
+        """
+        sp = self.spectral.sparse_lib
+        N = np.prod(self.init[0][1:])
+
+        Id = sp.eye(N)
+        Pl_lhs = {comp: {comp: Id} for comp in self.components}
+        self.Pl = self._setup_operator(Pl_lhs)
+
+        if left_preconditioner:
+            # reverse Kronecker product
+
+            if self.spectral.useGPU:
+                R = self.Pl.get().tolil() * 0
+            else:
+                R = self.Pl.tolil() * 0
+
+            for j in range(self.ncomponents):
+                for i in range(N):
+                    R[i * self.ncomponents + j, j * N + i] = 1.0
+
+            self.Pl = self.spectral.sparse_lib.csc_matrix(R)
+
+        if Dirichlet_recombination and type(self.axes[-1]).__name__ in ['ChebychevHelper, Ultraspherical']:
+            _Pr = self.spectral.get_Dirichlet_recombination_matrix(axis=-1)
+        else:
+            _Pr = Id
+
+        Pr_lhs = {comp: {comp: _Pr} for comp in self.components}
+        self.Pr = self._setup_operator(Pr_lhs) @ self.Pl.T
+
+    def solve_system(self, rhs, dt, u0=None, *args, skip_itransform=False, **kwargs):
+        """
+        Solve (M + dt*L)u=rhs. This requires that you setup the operators before using the functions ``GenericSpectralLinear.setup_L`` and ``GenericSpectralLinear.setup_M``. Note that the mass matrix need not be invertible, as long as (M + dt*L) is. This allows to solve some differential algebraic equations.
+
+        Note that in implicit Euler, the right hand side will be composed of the initial conditions. We don't want that in lines that don't depend on time. Therefore, we multiply the right hand side by the mass matrix. This means you can only do algebraic conditions that add up to zero. But you can easily overload this function with something more generic if needed.
+
+        We use a tau method to enforce boundary conditions in Chebychov methods. This means we replace a line in the system matrix by the polynomials evaluated at a boundary and put the value we want there in the rhs at the respective position. Since we have to do that in spectral space along only the axis we want to apply the boundary condition to, we transform back to real space after applying the mass matrix, and then transform only along one axis, apply the boundary conditions and transform back. Then we transform along all dimensions again. If you desire speed, you may wish to overload this function with something less generic that avoids a few transformations.
```

Comment:
Found on "chov"!

Reply:
I removed the need for the comment as well as the comment by implementing more efficient boundary conditions here.